src/NServiceBus.Core/Unicast/Messages/DeserializeLogicalMessagesConnector.cs

```diff
@@ -9,13 +9,21 @@
     using NServiceBus.Pipeline;
     using NServiceBus.Pipeline.Contexts;
     using NServiceBus.Scheduling.Messages;
-    using NServiceBus.Serialization;
+    using NServiceBus.Serializers;
+    using NServiceBus.Unicast;
     using NServiceBus.Unicast.Messages;
     using NServiceBus.Unicast.Transport;
 
     class DeserializeLogicalMessagesConnector : StageConnector<PhysicalMessageProcessingStageBehavior.Context, LogicalMessagesProcessingStageBehavior.Context>
     {
-        public IMessageSerializer MessageSerializer { get; set; }
+        readonly MessageDeserializerResolver deserializerResolver;
+
+        public DeserializeLogicalMessagesConnector(MessageDeserializerResolver deserializerResolver)
+        {
+            this.deserializerResolver = deserializerResolver;
+        }
+
+        public UnicastBus UnicastBus { get; set; }
```

Comment:
I think you don't need to test spring if you change your `MessageDeserializerResolver` design slightly. I'll try to comment on the file